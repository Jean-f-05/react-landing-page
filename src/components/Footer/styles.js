import styled, { css } from 'styled-components';
import { Wrapper as FooterElement } from '../../components/TextComponent/styles';
import { Wrapper as SectionContainer } from '../../components/SectionContainer/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
  text-align: center;
  border-top: 0.1rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration:none;
      font-size: ${theme.fonts.small};
      }

  & ${FooterElement} {
    font-size: ${theme.fonts.small};
}

  & ${SectionContainer}{
      padding-top: 0;
      padding-bottom: 0;
  }
`}`;
