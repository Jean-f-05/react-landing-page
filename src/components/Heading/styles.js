import styled, { css } from 'styled-components';

export const Header = styled.h1`
    color: ${({ theme, light }) =>
      css` ${light ? theme.colors.white : theme.colors.primaryColor}`}`;
