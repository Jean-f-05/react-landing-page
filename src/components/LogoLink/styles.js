import styled from 'styled-components';

export const Wrapper = styled.a`
    display: flex;
    align-items: center;
    text-decoration: none;
    color: inherit;

    > img {
        height: 3rem;
    }
`;
