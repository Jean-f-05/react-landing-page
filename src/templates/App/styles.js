import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
    ${({ theme, color }) => css`
    background-color: ${theme.colors.mainBg};
    color:${color}
    `}  
`;
