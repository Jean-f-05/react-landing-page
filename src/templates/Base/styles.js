import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
  padding-top: 5.4rem;

  @media ${theme.media.upToMedium} {
    padding-top: 0 ;
  }
`}`;
