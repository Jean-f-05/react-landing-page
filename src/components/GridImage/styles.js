import styled, { css } from 'styled-components';
import { Header as HeadingContainer } from '../Heading/styles';
import { Wrapper as TextComponent } from '../TextComponent/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
  & > ${TextComponent}{
    margin-bottom: ${theme.spacing.huge};
  }
`}`;

export const Grid = styled.div`
  ${({ theme }) => css`
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(28rem,1fr));
  gap:${theme.spacing.large}
`}`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
`}`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    transition: all 300ms ease-in-out;

    &:hover {
      transform: scale(1.2) rotate(15deg);
    }
`}`;
