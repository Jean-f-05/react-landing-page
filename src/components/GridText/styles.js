import styled, { css } from 'styled-components';
import { Header as HeadingContainer } from '../Heading/styles';
import { Wrapper as TextComponent } from '../TextComponent/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
  ${TextComponent}{
    margin-bottom: ${theme.spacing.huge};
  }
`}`;

export const Grid = styled.div`
  ${({ theme }) => css`
  counter-reset: grid-counter;
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(28rem,1fr));
  gap:${theme.spacing.large}
`}`;

export const GridElement = styled.div`
  ${({ theme }) => css`
  ${HeadingContainer}{
    position: relative;
    left: 5rem;
  }
  ${HeadingContainer}:before{
    counter-increment: grid-counter;
    content: counter(grid-counter);
    position: absolute;
    font-size: 7rem;
    top: -3rem;
    left: -5rem;
    transform: rotate(5deg);
  }
`}`;
