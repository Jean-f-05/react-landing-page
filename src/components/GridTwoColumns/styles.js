import styled, { css } from 'styled-components';
import { Header } from '../Heading/styles';

export const Wrapper = styled.div`
    ${({ theme, background }) => css`
        display: grid;
        grid-template-columns: 1fr 2fr;
        align-items: center;
        gap: ${theme.spacing.large};

        @media ${theme.media.upToMedium} {
            grid-template-columns: 1fr ;
            text-align: center;
        };

        ${Header}{
            margin-bottom:${theme.spacing.large};
        };
`}`;

export const TextWrapper = styled.div`
    ${({ theme }) => css`
    @media ${theme.media.upToMedium} {
        margin-bottom:${theme.spacing.large};
        `}
    }
`;

export const ImageWrapper = styled.div``;

export const Image = styled.img``;
