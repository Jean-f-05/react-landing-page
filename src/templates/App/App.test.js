import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';
import { theme } from '../../styles/theme';
import Home from '.';

test('renders learn react link', () => {
  renderTheme(<Home />);
  screen.logTestingPlaygroundURL();

  const headingContainer = screen.getByRole('heading', {
    name: /hello/i,
  }).parentElement;

  expect(headingContainer).toHaveStyle({
    backgroundColor: theme.colors.mainBg,
  });

  expect(headingContainer).toMatchSnapshot();
});
