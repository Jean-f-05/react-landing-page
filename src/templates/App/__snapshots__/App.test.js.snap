// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
.c0 {
  background-color: red;
  color: violet;
}

<div
  class="c0"
  color="violet"
>
  <h1>
    Hello
  </h1>
</div>
`;
